sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_buffered_producer_async.py

```diff
@@ -490,3 +490,51 @@ async def on_error(events, pid, err):
 
     await consumer.close()
     await receive_thread
+
+@pytest.mark.liveTest
+@pytest.mark.asyncio
+async def test_long_wait_small_buffer(connection_str):
```

Comment:
Did this test actually repro the reported hanging bug before we fixed it?

Reply:
No this was a test I added in after the fact, this was the repro code to catch the bug.